druid-shell/src/platform/x11/window.rs

```diff
@@ -83,39 +89,111 @@ impl WindowBuilder {
         // TODO(x11/menus): implement WindowBuilder::set_menu (currently a no-op)
     }
 
+    fn atoms(&self, id: u32) -> Result<WindowAtoms, Error> {
+        let conn = self.app.connection();
+
+        let wm_protocols = match xcb::intern_atom(conn, false, "WM_PROTOCOLS").get_reply() {
+            Ok(reply) => reply.atom(),
+            Err(err) => {
+                return Err(Error::Generic(format!(
+                    "failed to get WM_PROTOCOLS: {}",
+                    err
+                )))
+            }
+        };
+
+        let wm_delete_window = match xcb::intern_atom(conn, false, "WM_DELETE_WINDOW").get_reply() {
+            Ok(reply) => reply.atom(),
+            Err(err) => {
+                return Err(Error::Generic(format!(
+                    "failed to get WM_DELETE_WINDOW: {}",
+                    err
+                )))
+            }
+        };
+
+        let protocols = [wm_delete_window];
+        // TODO(x11/errors): Check the response for errors?
+        xcb::change_property(
+            conn,
+            PROP_MODE_REPLACE as u8,
+            id,
+            wm_protocols,
+            ATOM_ATOM,
+            32,
+            &protocols,
+        );
```

Comment:
It's the wonders of X11. :smile: It registers the window for window closing messages. X11 doesn't actually have a window closing message itself, because it doesn't have a close button. Various window managers however add a close button and then send a close message via an X11 extension.

So this sets a property for our X11 window, saying that we are interested in the `WM_DELETE_WINDOW` event by having that be part of our `WM_PROTOCOLS` array. Both of these are extensions to X11.

I'll probably add some sort of inline comment to describe as much.